feat(worktree): add --dir flag to decouple folder names from branches

webjunkie opened the pull request:
## Summary
- Adds optional `--dir <dirname>` parameter to `create`, `checkout`, and `pr` commands
- Decouples worktree directory names from branch names
- Enables theme-based workflows where multiple related branches can be worked on in the same worktree

## Motivation
Previously, worktree folder names were always tied to branch names (1:1 relationship). For multi-PR workflows around a theme, this meant creating new worktrees for each branch, losing context and setup.

With `--dir`, you can:
```bash
posthog-worktree create first-feature --dir auth-theme
cd ~/.worktrees/posthog/auth-theme
# work, commit, create PR

git checkout -b second-feature
# work on related feature in same worktree, maintaining Claude context
```

## Changes
- Rewrote argument parsing to support `--dir` flag alongside positional args
- Updated `create_worktree()`, `checkout_worktree()`, and `checkout_pr_worktree()` to use custom dirname when provided
- Updated help text with examples
- `remove` command unchanged (already finds worktrees by branch name)

## Test plan
- [x] Tested `create` with `--dir` flag
- [x] Verified worktree created at custom path with correct branch
- [x] Test `checkout` with `--dir`
- [x] Test `pr` with `--dir`
- [ ] Test that existing workflows without `--dir` still work

webjunkie committed:
feat(worktree): add --dir flag to decouple folder names from branches

Adds optional --dir parameter to create, checkout, and pr commands. Enables theme-based workflows where multiple related branches can be worked on in the same worktree directory while maintaining separate branch names.

Previously, worktree folder names were always tied to branch names, forcing a 1:1 relationship. Now you can specify a custom directory name that persists across branch switches within that worktree.

haacked reviewed (approved):
Nice! I think the issue that Greptile points out is legit, but not a blocker.

webjunkie committed:
fix(worktree): validate --dir flag argument exists

Adds validation to ensure --dir flag receives an argument. Prevents errors when --dir is used without a value.
